Lesson10/src/Shop.java

```diff
@@ -0,0 +1,82 @@
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Shop {
+    List<Product> shop = new LinkedList<>();
+    Scanner scanner = new Scanner(System.in);
+
+    public List<Product> getShop() {
+        return shop;
+    }
+
+    public void sortPrice(Shop shop) {
+        List<Product> sortListPrice = new LinkedList<>(shop.getShop());
+        sortListPrice.sort(Comparator.comparing(Product::getPrice).reversed());
+        System.out.println("Список всех товаров, сортированный по цене:");
+        for (Product item : sortListPrice) {
+            System.out.println(item.getName() + ". Цена: " + item.getPrice());
+        }
+    }
+
+    public void addProduct(Product product) throws ProductAlreadyExist {
+        for (int i = 0; i < shop.size(); i++) {
+            if (product.getId() == shop.get(i).getId()) {
+                throw new ProductAlreadyExist("Товар с таким id уже существует");
+            } else shop.add(product);
+            break;
+        }
+    }
```

Comment:
you will get a ConcurrentModificationException here, because you are trying to add element to the collection that you are iterating through

First you need to go through list and check that there is no element with given id, then, after for loop, you can add new element